uint64 nanoseconds
string source
string message_name
uint32 message_number
string message
